database/prefixdb/db.go

```diff
@@ -233,6 +238,21 @@ func (db *Database) prefix(key []byte) *[]byte {
 	return prefixedKey
 }
 
+func (db *Database) suffix() *[]byte {
+	keyLen := len(db.dbPrefix)
+	prefixedKey := db.bufferPool.Get(keyLen)
+	copy(*prefixedKey, db.dbPrefix)
+	n := len(db.dbPrefix)
+	// lexically increment prefix to find the suffix of this db range
+	for i := n - 1; i >= 0; i-- {
+		(*prefixedKey)[i]++
+		if (*prefixedKey)[i] != 0 {
+			break
+		}
+	}
+	return prefixedKey
+}
```

Comment:
Would it make sense to define `dbPrefix` as a constant? Similar to `dbPrefix`? I don't think it can ever change... So we probably don't need to allocate a new slice for it whenever we need it.

Reply:
You mean dbSuffix? And as a member of the struct, not a constant?

It's only ever used by Compact(), as there are no iterators that define a limit in the API. So, maybe?

Reply:
ok sure, pushing